### ISSUE TITLE
All data Bars become invisible to avoid overlap (time chart)

### ISSUE THREAD

**Issue by Fresh-Cut-Grass** [recharts/recharts#2758]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/condescending-visvesvaraya-hz0c3)

### Steps to reproduce
1) Create a datetime BarChart 
2) Give it a large enough domain
3) Make some of the data really close to one another compared to the whole domain

### What is expected?
The data bars that aren't overlapping should remain visible (be 1px wide at the very least).

Or the overlapping data should be visible (and overlapping), especially since I'm "forcing" a value using `barSize`.

### What is actually happening?
**All** the bars become invisible (`barSize` seems to be 0). Even the ones that aren't overlapping.

| Environment | Info |
|---|---|
| Recharts | v2.1.6 |
| React | react 17.0.2 |
| System | Windows 10 |
| Browser | Chrome |

---
This is the "normal" behaviour but I feel like the min barSize should always be bigger than 0.

I would expect either the bars to overlap (since I'm specifying a value for `barSize`, after all) or, at the very least, be at least 1px wide. The most important part would be for the other data (not overlapping) to remain visible.

This computed size is applied even to the bars that aren't overlapping. It would be nice if there was a `minBarSize` property that would be affected to the overlapping bars while the other data bars continue to respect the `barSize` property (or whatever value ReCharts computed that isn't 0 and is bigger than `minBarSize`).

(optional) Feature request: 
Ideally, a data-grouping functionality (like HighCharts offers: "Data grouping replaces a sequence of data points in a series with one grouped point. The values of each grouped point is calculated from the original values of every point used." https://www.highcharts.com/demo/stock/data-grouping) would be perfect, since the user can then Zoom in and view the details.

I will try and do this myself by pre-processing the data.

**Comment by Fresh-Cut-Grass** [comment]
Quick update: 
since I haven't managed to smartly group the data yet, I went ahead and used the `shape` property of `Bar` to draw a `<rect />` with a min width of 1px.

As I suspected, the Bars weren't invisible but, instead, had a width of `0.0000713...` making them invisible. On top of that, you use a `path` to draw the Bars, making them harder to edit (the computed width is used in the path). Using 

```js
<Bar
  shape={({ fill, x, y, width, height, ...props }) => {
    return (
      <rect
        x={x}
        y={y}
        width={Math.max(1, width)}
        height={height}
        fill={fill}
      />
    );
  }}
/>
```

Despite this temporary fix, I still think ReCharts shouldn't give a width of less than `1px`. Since it can be fixed with the `shape` attribute, the default behaviour could remain the same, but adding a `minBarSize` property could be neat.

Just to be clear: 
- my "solution" DOES create overlapping Bars
- the behaviour I had initially explained isn't a bug after all: the Bars just had a really small width (`0.0000713...`). Feel free to close this issue (since it is now turning into a feature request, kind of)

**Comment by etothepii** [comment]
I agree it is a bug, since this is the behaviour I would expect from barSize.

However, I don't pay for recharts and its amazing, so no complaints!

**Comment by Irtiza-h30** [comment]
+1

**Comment by bobbeugels** [comment]
Ran into the same problem, barSize is doing nothing for me. Made it work by passing a stroke color and a stroke width.

**Comment by ckifer** [comment]
bar charts on a time X Axis (or bar charts with a numeric domain) are not really supported at the moment. Changing the XAxis to be categorical allows everything to render. Needs looked into more in order to figure out how to support numeric bar charts but its generally not the best idea from a viz perspective..
